soc/riscv/esp32c3/Kconfig.soc

```diff
@@ -26,14 +26,6 @@ config ESPTOOLPY_FLASHFREQ_80M
 	bool
 	default y
 
-config FLASH_SIZE
-	int
-	default $(dt_node_reg_size_int,/soc/flash-controller@60002000/flash@0,0)
```

Comment:
There have some CI failure caused by esp32c3 not set `unit`, but `FLASH_SIZE` in `arch/Kconfig` set `uint` to `K`.
```
arch/Kconfig:
  default $(dt_chosen_reg_size_int,$(DT_CHOSEN_Z_FLASH),0,K) if (XIP && (ARM ||ARM64)) || !ARM

esp32c3/Kconfig.soc:
  default $(dt_node_reg_size_int,/soc/flash-controller@60002000/flash@0,0)
```
I think we need to keep the `FLASH_SIZE` and `FLASH_BASE_ADDRESS` config for esp32c3.

Reply:
done!

Reply:
@cameled @ottojo, thanks for keeping ESP32-C3 as is, otherwise it would not work.